roles/prometheus/tasks/cleanup.yml

```diff
@@ -1,8 +1,15 @@
 # code: language=ansible
 ---
+- name: Check the status of firewalld
+  ansible.builtin.systemd:
+    name: firewalld.service
+  register: firewalld_status
+
 - name: Close prometheus port using firewalld
   ansible.posix.firewalld:
     port: "{{ prometheus_port }}/tcp"
     permanent: true
     state: disabled
     immediate: true
+    offline: true
+  when: firewalld_status.status.ActiveState == "active"
```

Comment:
what's the meaning of `active`?
`offline` is used to set the configuration in moments where firewalld is not running.

Reply:
yeah, now that I think about it, it's a bit pointless to have the offline: true, because the when clause is already checking if it's running (and it won't execute if it isn't running).

the `active` value belongs to the firewalld status fact we collect previously, this is a way of preventing the execution that doesn't require firewalld to be there using the systemd_module as suggested by @stefanotorresi

Reply:
but this is the prometheus host...
it could still run without firewalld I guess.
what we need to check is if firewalld is installed, and in this case, with the `offline: true` set the exceptions

Reply:
Ok, I'll switch to the zypper module